src/TestExplorer/TestParsers/SwiftTestingOutputParser.ts

```diff
@@ -155,31 +176,131 @@ export class SwiftTestingOutputParser {
         return !matches ? id : matches[1];
     }
 
+    private testCaseId(testId: string, testCaseId: string): string {
+        const testCase = this.testCaseMap.get(testId)?.get(testCaseId);
+        return testCase ? this.createTestCaseId(testCase) : testId;
+    }
+
+    // Test cases do not have a unique ID if their arguments are not serializable
+    // with Codable. If they aren't, their id appears as `argumentIDs: nil`, and we
+    // fall back to using the testCase display name as the test case ID. This isn't
+    // ideal because its possible to have multiple test cases with the same display name,
+    // but until we have a better solution for identifying test cases it will have to do.
+    // SEE: rdar://119522099.
+    private createTestCaseId(testCase: TestCase): string {
+        return testCase.id === "argumentIDs: nil" ? testCase.displayName : testCase.id;
```

Comment:
question: Where does the "argumentIDs: nil" constant come from?

suggestion: Maybe add a comment so we know where we can check to make sure it is in sync

Reply:
This is the ID string for parameterized test cases that have unserializable arguments. Its a bit odd for sure, and liable to change in future swift-testing releases. The comment on this function describes the reason.